core/templates/pages/donate-page/donate-page.component.ts

```diff
@@ -59,10 +59,16 @@ export class DonatePageComponent implements OnInit, OnDestroy {
     private translateService: TranslateService,
     private alertsService: AlertsService,
     private mailingListBackendApiService: MailingListBackendApiService,
-    private ngbModal: NgbModal
+    private ngbModal: NgbModal,
+    private renderer2: Renderer2
   ) {}
 
   ngOnInit(): void {
+    const scriptElement = document.createElement('script');
+    scriptElement.src = 'https://donorbox.org/widget.js';
+    scriptElement.setAttribute('paypalExpress', 'false');
+    scriptElement.async = true;
+    this.renderer2.appendChild(document.body, scriptElement);
```

Comment:
This is perfect. One last request. Can you move this into a service. Here are the requirements for that service:

1. The service should contain an enum with properties like: `donorBox`. You can call the enum KNOWN_SCRIPTS or APPROVED_SCRIPTS.
2. The service exposes a function called loadScript that takes only a value from the enum as input.
3. There are also some scripts that fail if they are loaded twice. I not sure if the donor box script will have such a painful problem or not. Better to be safe than sorry. Add a check to make sure that a script that has been loaded isn't fetched and loaded again.

One other thing that we would need to confirm is if using renderer removes the script on destroy. If that is the case, then lets use the document instead of renderer. We want to just fetch these scripts only once and not have to fetch them again when the use comes back to donate page. Might be an unlikely scenario but this can be useful for other scripts we have.

Reply:
Also set oppia/core-reviewers and oppia/frontend-infrastructure-reviewers as the owners.

Reply:
done.